### ISSUE TITLE
ESPHome: Cannot connect to MQTT TLS esp-tls: 0x800b

### ISSUE THREAD

**Issue by mrkeuz** [mrkeuz/esphome-mqtt-tls#1]
Hi @ShellAddicted, 

I'm testing [2022.6.2](https://github.com/esphome/esphome/releases/tag/2022.6.2) release and try connect to  MQTT TLS. But without luck. I prepare sample [project](https://github.com/mrkeuz/esphome-mqtt-tls) for testing. Could you review config?

There is a [public mqtt broker](https://test.mosquitto.org/) from mosquitto: 

- `test.mosquitto.org:8883` - does not require user/password authentication

I successfully connect from desktop, but cannot connect from `ESPhome`. Certificate valid (I check it [here](https://www.sslshopper.com/certificate-decoder.html)) see [ISRG_root_x1.pem](https://github.com/mrkeuz/esphome-mqtt-tls/blob/0ea5667fb75c794ea62cec4800339d1fde19504c/ISRG_root_x1.pem) in project root. 

What am I doing wrong here?

Logs from `esp32`:

```log
...
[04:42:28][I][mqtt:176]: Connecting to MQTT...
[04:42:28][D][esp-idf:000]: E (5804) esp-tls-mbedtls: mbedtls_x509_crt_parse returned -0x2780
[04:42:28]
[04:42:28][D][esp-idf:000]: E (5805) esp-tls-mbedtls: Failed to set client configurations
[04:42:28]
[04:42:28][D][esp-idf:000]: E (5805) esp-tls: create_ssl_handle failed
[04:42:28]
[04:42:28][D][esp-idf:000]: E (5806) esp-tls: Failed to open new connection
[04:42:28]
[04:42:28][D][esp-idf:000]: E (5813) TRANS_SSL: Failed to open a new connection
[04:42:28]
[04:42:28][D][esp-idf:000]: E (5815) MQTT_CLIENT: Error transport connect
[04:42:28]
[04:42:28][E][mqtt.idf:119]: MQTT_EVENT_ERROR
[04:42:28][E][mqtt.idf:121]: Last error code reported from esp-tls: 0x800b
[04:42:28][E][mqtt.idf:122]: Last tls stack error number: 0x2780
[04:42:28][E][mqtt.idf:124]: Last captured errno : 0 (Success)
[04:42:28][W][mqtt:264]: MQTT Disconnected: TCP disconnected.
...
```

Related to https://github.com/esphome/esphome/pull/3580

**Comment by ShellAddicted** [comment]
Don't use `!include` AFAIK that's only for YAML files.
Paste the CA file directly in the yaml config like shown in the docs
https://esphome.io/components/mqtt.html#tls-with-esp-idf-esp32

Also, looks like the `8883` port has a different CA than Let's Encrypt, download the CA file directly from test.mosquitto.org

**Comment by mrkeuz** [comment]
Thank you very much! 

Finally, got it working! Fixed in [c8a7ee8](https://github.com/mrkeuz/esphome-mqtt-tls/commit/c8a7ee8ed97961cbe474a5bd993bf9f44242b35b) 

Certificate passed from https://test.mosquitto.org/ssl/mosquitto.org.crt. 

Directly exported certificate (from browser from `test.mosquitto.org:8883`) different and don't pass. By some reason. Strange, but not important. And yes seems `!include` does not work in this case. 

The main thing is that it worked at least in some form.

Thanks! 🚀🚀🚀

**Comment by mrkeuz** [comment]
FYI: Let's encrypt chain finally also working (with ISRG_root_x1)

Many, many thanks for help!
